### PR TITLE
rsx: Use bitfield template to decode values.

### PR DESCRIPTION
**Pull request by vlj**
This PR reverts the explicit assignment of member in rsx_state. It uses bitfield class instead of explicit << and &.

### CODE REVIEW

**Comment by raven02** [comment]
fix https://github.com/RPCS3/rpcs3/issues/1944 .howver something seems to be broken in-game

untitled

**Comment by raven02** [comment]
Disgaea 4 hits this

F {rsx::thread} class fmt::exception thrown: rsx::blit_engine::to_context_dma(): Unknown context dma 0x0
(in file C:\rpcs3\rpcs3\Emu\RSX\GCM.cpp:1702)

**Comment by vlj** [comment]
This is main menu or ingame ?

**Comment by raven02** [comment]
It is the title menu. Should look like this

untitled

**Comment by SakataGintokiYT** [comment]
-Naruto Games 
-JoJo's Bizarre Adventure: All Star Battle
-Dragon Ball: Raging Blast 1
-Initial D Extreme Stage
F {rsx::thread} class fmt::exception thrown: rsx::blit_engine::to_context_dma(): Unknown context dma 0x0
(in file C:\rpcs3\rpcs3\Emu\RSX\GCM.cpp:1702)

File: rpcs3/Emu/RSX/rsx_methods.h [hunk 1]
```diff
@@ -111,1176 +122,958 @@ namespace rsx
...
 		u16 vertex_attrib_input_mask() const
 		{
-			return m_vertex_attrib_input_mask;
+			return decode<NV4097_SET_VERTEX_ATTRIB_INPUT_MASK>().mask();
 		}
 
 		u16 frequency_divider_operation_mask() const
 		{
-			return m_frequency_divider_operation_mask;
+			return decode<NV4097_SET_FREQUENCY_DIVIDER_OPERATION>().m_frequency_divider_operation_mask();
```
Comment (raven02) [inline rpcs3/Emu/RSX/rsx_methods.h]: Probably decode<NV4097_SET_FREQUENCY_DIVIDER_OPERATION>().frequency_divider_operation_mask();

**Comment by vlj** [comment]
Disgaea3 should be fixed

**Comment by raven02** [comment]
Thank @vlj

**Comment by raven02** [comment]
Add default value for context dma report should fix the above error .

`		registers[NV4097_SET_CONTEXT_DMA_REPORT] = CELL_GCM_CONTEXT_DMA_TO_MEMORY_GET_REPORT;`

**Comment by vlj** [comment]
The context dma crash wasn't there before ?

**Comment by SakataGintokiYT** [comment]
@vlj no

**Comment by raven02** [comment]
@vlj . yes , it didn't happen before until this commit.

**Pull request merged**